pkg/ipoib/ipoib.go

```diff
@@ -47,38 +93,41 @@ func (im *ipoibManager) CreateIpoibLink(conf *types.NetConf, ifName string, netn
 		Umcast: 1,
 	}
 
-	if err := netlink.LinkAdd(ipoibLink); err != nil {
+	if err := im.nLink.LinkAdd(ipoibLink); err != nil {
 		return nil, fmt.Errorf("failed to create interface: %v", err)
 	}
-	link, err := netlink.LinkByName(tmpName)
+	link, err := im.nLink.LinkByName(tmpName)
 	if err != nil {
 		return nil, err
 	}
 
-	if err = netlink.LinkSetNsFd(link, int(netns.Fd())); err != nil {
+	if err = im.nLink.LinkSetNsFd(link, int(netns.Fd())); err != nil {
 		return nil, fmt.Errorf("failed to move interfaceee %s to netns: %v", tmpName, err)
 	}
 
 	err = netns.Do(func(_ ns.NetNS) error {
 		ipv4SysctlValueName := fmt.Sprintf(ipV4InterfaceArpProxySysctlTemplate, tmpName)
-		if _, err := sysctl.Sysctl(ipv4SysctlValueName, "1"); err != nil {
+		if _, err := im.nLink.SetSysVal(ipv4SysctlValueName, "1"); err != nil {
 			// remove the newly added link and ignore errors, because we already are in a failed state
-			_ = netlink.LinkDel(ipoibLink)
+			_ = im.nLink.LinkDel(ipoibLink)
 			return fmt.Errorf("failed to set proxy_arp on newly added interface %q: %v", tmpName, err)
 		}
 
-		err := ip.RenameLink(tmpName, ifName)
-		if err != nil {
-			_ = netlink.LinkDel(ipoibLink)
+		if err := im.nLink.LinkSetDown(link); err != nil {
+			return err
+		}
+		if err := im.nLink.LinkSetName(link, ifName); err != nil {
+			_ = im.nLink.LinkDel(ipoibLink)
 			return fmt.Errorf("failed to rename interface to %q: %v", ifName, err)
 		}
+		if err := im.nLink.LinkSetUp(link); err != nil {
+			return err
+		}
 		iface.Name = ifName
 
-		ipoibContLink, err := netlink.LinkByName(ifName)
 		if err != nil {
```

Comment:
check error of what? you remove the "ipoibContLink, err := netlink.LinkByName(ifName)"

Reply:
by mistake